hawkmoth/__init__.py

```diff
@@ -171,7 +177,16 @@ def run(self):
 
         for filename in filenames:
             self.__build_cache(cache, clang_args, filename)
-            self.__get_docstrings(cache, clang_args, result, filename)
+            if self.__get_docstrings(cache, clang_args, result, filename):
+                matches.append(filename)
+
+        nmatches = len(matches)
+        if nmatches == 0:
+            self.logger.warning('No matching docstring.',
+                                location=(self.env.docname, self.lineno))
+        elif nmatches > 1:
```

Comment:
Bah, this needs to be `auto and ...` or it would trigger in the `cautodoc` directive when issued with globs or multiple files.